x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/index.tsx

```diff
@@ -399,6 +405,25 @@ export function Detail() {
         pkgkey,
       });
 
+      /** Users from Security Solution onboarding page will have onboardingLink and onboardingAppId in the query params
+       ** to redirect back to the onboarding page after adding an integration
+       */
+      const navigateOptions: InstallPkgRouteOptions =
+        onboardingAppId && onboardingLink
+          ? [
+              defaultNavigateOptions[0],
+              {
+                ...defaultNavigateOptions[1],
+                state: {
+                  ...(defaultNavigateOptions[1]?.state ?? {}),
+                  onCancelNavigateTo: [onboardingAppId, { path: onboardingLink }],
+                  onCancelUrl: onboardingLink,
+                  onSaveNavigateTo: [onboardingAppId, { path: onboardingLink }],
```

Comment:
Back to given page if onboardingAppId && onboardingLink are provided on integration saved or cancelled.